packages/ember-metal/lib/utils.js

```diff
@@ -280,35 +270,45 @@ export function guidFor(obj) {
   @param {Function} superFunc The super function.
   @return {Function} wrapped function.
 */
-
-export function wrap(func, superFunc) {
-  function superWrapper() {
-    var ret;
-    var sup  = this && this.__nextSuper;
-    var length = arguments.length;
-
-    if (this) {
-      this.__nextSuper = superFunc;
+export function wrap(func, _superFunc) {
+  var superFunc = _superFunc;
+  var hasSuper;
+  if (sourceAvailable) {
+    hasSuper = func.__hasSuper;
+
+    if (hasSuper === undefined) {
+      hasSuper = func.toString().indexOf('_super') > -1;
+      func.__hasSuper = hasSuper;
```

Comment:
The paranoid me wants to suggest leaving `__nextSuper` and the `enumerable: false` and using that instead of a new property. I know you probably want to drop the `defineProperty`, but it does not seem to be a change core to what you've done in this PR.